cocos/editor-support/cocostudio/ActionTimeline/CCActionTimeline.cpp

```diff
@@ -355,6 +373,52 @@ void ActionTimeline::emitFrameEvent(Frame* frame)
     }
 }
 
+void ActionTimeline::addFrameEndCallFunc(int frameIndex, const std::string& funcKey, std::function<void()> func)
+{
+    if (func != nullptr)
+    {
+        _frameEndCallFuncs[frameIndex][funcKey] = func;
+    }
+}
+
+void ActionTimeline::removeFrameEndCall(int frameIndex, const std::string& funcKey)
+{
+    auto endClipCallsIter = _frameEndCallFuncs.find(frameIndex);
+    if (endClipCallsIter != _frameEndCallFuncs.end())
+    {
+        auto funcIter = (*endClipCallsIter).second.find(funcKey);
+        if (funcIter != (*endClipCallsIter).second.end())
+            (*endClipCallsIter).second.erase(funcKey);
+        if ((*endClipCallsIter).second.empty())
+            _frameEndCallFuncs.erase(endClipCallsIter);
+    }
+}
+
+void ActionTimeline::removeFrameEndCall(int frameIndex)
+{
+    auto endClipCallsIter = _frameEndCallFuncs.find(frameIndex);
+    if (endClipCallsIter != _frameEndCallFuncs.end())
+    {
+        _frameEndCallFuncs.erase(endClipCallsIter);
+    }
+}
+
+void ActionTimeline::clearFrameEndCalls()
```

Comment:
Please keep API name consistency: rename `ActionTimeline::clearFrameEndCalls` to `ActionTimeline::clearFrameEndCallFuncs`